### PR TITLE
First stage

### PR DESCRIPTION
**Pull request by monika1127**

### CODE REVIEW

File: .eslintcache [hunk 1]
```diff
@@ -0,0 +1 @@
...{"ruleId":"51","replacedBy":"52"},{"ruleId":"53","replacedBy":"54"},{"ruleId":"55","severity":1,"message":"56","line":18,"column":76,"nodeType":"57","messageId":"58","endLine":18,"endColumn":78},{"ruleId":"55","severity":1,"message":"59","line":25,"column":17,"nodeType":"57","messageId":"58","endLine":25,"endColumn":28},{"ruleId":"55","severity":1,"message":"60","line":25,"column":44,"nodeType":"57","messageId":"58","endLine":25,"endColumn":47},{"ruleId":"55","severity":1,"message":"61","line":26,"column":17,"nodeType":"57","messageId":"58","endLine":26,"endColumn":23},{"ruleId":"55","severity":1,"message":"62","line":26,"column":68,"nodeType":"57","messageId":"58","endLine":26,"endColumn":74},{"ruleId":"55","severity":1,"message":"56","line":26,"column":76,"nodeType":"57","messageId":"58","endLine":26,"endColumn":78},"no-native-reassign",["63"],"no-negated-in-lhs",["64"],"no-unused-vars","'id' is assigned a value but never used.","Identifier","unusedVar","'companyName' is assigned a value but never used.","'tel' is assigned a value but never used.","'number' is assigned a value but never used.","'isPaid' is assigned a value but never used.","no-global-assign","no-unsafe-negation"]
```
Comment (korczas) [inline .eslintcache]: add `.eslintcache` to your `.gitignore` file

File: db.json [hunk 2]
```diff
@@ -1,56 +1,63 @@
...
+    {
+      "id": 3,
+      "companyName": "Prosiaczek i Przyjaciele",
+      "NIP": "556-786-553-888",
+      "street": "Liliowa 3",
+      "post": "55-888",
+      "city": "Stumilowy Las",
+      "tel": "999-93-29"
+    }
+  ]
```
Comment (korczas) [inline db.json:62]: add empty line on end of file

File: package.json [hunk 3]
```diff
@@ -6,6 +6,7 @@
     "@testing-library/jest-dom": "^5.11.8",
     "@testing-library/react": "^11.2.2",
     "@testing-library/user-event": "^12.6.0",
+    "cliui": "^7.0.4",
```
Comment (korczas) [inline package.json]: `cliui` library seems to be redundant in your app.\
run `npm uninstall cliui`

File: public/index.html [hunk 4]
```diff
@@ -15,6 +15,10 @@
       user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
     -->
     <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+    <script
+      src="https://kit.fontawesome.com/1981fa3dd6.js"
+      crossorigin="anonymous"
+    ></script>
```
Comment (korczas) [inline public/index.html]: font awesome is an old school. Use .svg instead

File: src/App.css [hunk 5]
```diff
@@ -1,38 +1,12 @@
...
-@keyframes App-logo-spin {
-  from {
-    transform: rotate(0deg);
-  }
-  to {
-    transform: rotate(360deg);
-  }
+.App {
+  text-align: center;
+  font-size: 12px;
```
Comment (korczas) [inline src/App.css]: your body font-size is set for 12px, so duplicating it here seems redundant

File: src/assets/images/clock2.min.svg [hunk 6]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="32" height="32"><path d="M16 0C7.163 0 0 7.163 0 16s7.163 16 16 16 16-7.163 16-16S24.837 0 16 0zm4.586 23.414L14 16.828V8h4v7.172l5.414 5.414-2.829 2.829z"/></svg>
```
Comment (korczas) [inline src/assets/images/clock2.min.svg]: add empty new line at the end of file

File: src/assets/images/cog.svg [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<!-- Generated by IcoMoon.io -->
+<svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 32 32" fill='white'>
+    <title>cog</title>
+    <path
+        d="M29.181 19.070c-1.679-2.908-0.669-6.634 2.255-8.328l-3.145-5.447c-0.898 0.527-1.943 0.829-3.058 0.829-3.361 0-6.085-2.742-6.085-6.125h-6.289c0.008 1.044-0.252 2.103-0.811 3.070-1.679 2.908-5.411 3.897-8.339 2.211l-3.144 5.447c0.905 0.515 1.689 1.268 2.246 2.234 1.676 2.903 0.672 6.623-2.241 8.319l3.145 5.447c0.895-0.522 1.935-0.82 3.044-0.82 3.35 0 6.067 2.725 6.084 6.092h6.289c-0.003-1.034 0.259-2.080 0.811-3.038 1.676-2.903 5.399-3.894 8.325-2.219l3.145-5.447c-0.899-0.515-1.678-1.266-2.232-2.226zM16 22.479c-3.578 0-6.479-2.901-6.479-6.479s2.901-6.479 6.479-6.479c3.578 0 6.479 2.901 6.479 6.479s-2.901 6.479-6.479 6.479z">
+    </path>
+</svg>
```
Comment (korczas) [inline src/assets/images/cog.svg:7]: add empty new line at the end of file

File: src/assets/images/printer-black.svg [hunk 8]
```diff
@@ -0,0 +1,8 @@
+<!-- Generated by IcoMoon.io -->
+<svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 32 32">
+    <title>printer</title>
+    <path d="M8 2h16v4h-16v-4z"></path>
+    <path
+        d="M30 8h-28c-1.1 0-2 0.9-2 2v10c0 1.1 0.9 2 2 2h6v8h16v-8h6c1.1 0 2-0.9 2-2v-10c0-1.1-0.9-2-2-2zM4 14c-1.105 0-2-0.895-2-2s0.895-2 2-2 2 0.895 2 2-0.895 2-2 2zM22 28h-12v-10h12v10z">
+    </path>
+</svg>
```
Comment (korczas) [inline src/assets/images/printer-black.svg:8]: add empty new line at the end of file

File: src/assets/images/printer.svg [hunk 9]
```diff
@@ -0,0 +1,8 @@
+<!-- Generated by IcoMoon.io -->
+<svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 32 32" fill='white'>
+    <title>printer</title>
+    <path d="M8 2h16v4h-16v-4z"></path>
+    <path
+        d="M30 8h-28c-1.1 0-2 0.9-2 2v10c0 1.1 0.9 2 2 2h6v8h16v-8h6c1.1 0 2-0.9 2-2v-10c0-1.1-0.9-2-2-2zM4 14c-1.105 0-2-0.895-2-2s0.895-2 2-2 2 0.895 2 2-0.895 2-2 2zM22 28h-12v-10h12v10z">
+    </path>
+</svg>
```
Comment (korczas) [inline src/assets/images/printer.svg:8]: add empty new line at the end of file

File: package.json (hunk 3 above)
Comment (korczas) [reply package.json]: I assume you had installed `json-server` globally. Please add `json-server` to your local dependencies.
`npm i json-server`

File: src/components/invoices/InvoiceItem.js [hunk 10]
```diff
@@ -0,0 +1,56 @@
+import React, { Component } from 'react'
+//imported SVG files
+import { ReactComponent as Search } from '../../assets/images/search.svg'
+import { ReactComponent as Clock } from '../../assets/images/clock2.svg'
+import { ReactComponent as Printer } from '../../assets/images/printer-black.svg'
+import { ReactComponent as Arrow } from '../../assets/images/redo2.svg'
+
+import Button from '../layout/button/Button'
+
+import PropTypes from 'prop-types'
```
Comment (korczas) [inline src/components/invoices/InvoiceItem.js]: good practice for imports is to group them by type. Import libraries first, empty line, components, empty line, icons, empty line, styles, empty line

```
import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Button from '../layout/button/Button'

import { ReactComponent as Search } from '../../assets/images/search.svg'
import { ReactComponent as Clock } from '../../assets/images/clock2.svg'
import { ReactComponent as Printer } from '../../assets/images/printer-black.svg'
import { ReactComponent as Arrow } from '../../assets/images/redo2.svg'
```

File: src/components/layout/Spinner/Spinner.js [hunk 11]
```diff
@@ -0,0 +1,6 @@
+import './spinner.css'
+import React from 'react'
+
+const Spinner = () => <div className="loader">Loading...</div>
+
+export default Spinner
```
Comment (korczas) [inline src/components/layout/Spinner/Spinner.js]: new line

File: src/components/invoices/InvoiceDetails/InvoiceDetails.js [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+import PropTypes from 'prop-types'
+import TopBar from '../../layout/topBar/TopBar'
+import Button from '../../layout/button/Button'
+import Table from '../../layout/table/Table'
+
+import './invoiceDetails.css'
+
+import { ReactComponent as Mail } from '../../../assets/images/mail.svg'
+import { ReactComponent as Print } from '../../../assets/images/printer.svg'
+import { ReactComponent as Set } from '../../../assets/images/cog.svg'
```
Comment (korczas) [inline src/components/invoices/InvoiceDetails/InvoiceDetails.js]: group imports

File: src/components/invoices/InvoiceDetails/InvoiceDetails.js [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+export default class InvoiceDetails extends Component {
+
+    static propTypes = {
+        contractorInfo: PropTypes.object.isRequired,
+        invoice: PropTypes.object.isRequired,
+        closeInvoiceDetails: PropTypes.func.isRequired,
+    }
+
+
+
```
Comment (korczas) [inline src/components/invoices/InvoiceDetails/InvoiceDetails.js]: ✂️

File: src/components/invoices/InvoiceDetails/InvoiceDetails.js [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+        //dane pomocnicze -które powinny się zaciągać z jakiś baz danych których nie mam //
+        const VAT = 0.23
+        const showVAT = (VAT * 100) + '%'
+        const netto = Number.parseFloat(price / (1 + VAT)).toFixed(2)
+
+        const rabat = 0
+        const poRabacie = Number.parseFloat(netto - rabat).toFixed(2)
+
```
Comment (korczas) [inline src/components/invoices/InvoiceDetails/InvoiceDetails.js]: assume that `price` is netto, and calculate brutto based on this

File: src/components/invoices/InvoicesList.js [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+        invoices: [],
+        loading: false,
+        current: null,
+        contractorInfo: null
+    }
+
+    componentDidMount() {
+        this.setState({ loading: true })
+        fetch('http://localhost:5000/invoices')
+            .then(res => res.json())
```
Comment (korczas) [inline src/components/invoices/InvoicesList.js]: you can skip this `this.setState({ loading: true })` in cDM - that makes unnecessary render. Set you default state to `{..., loading: true, ...}`

File: src/components/layout/table/Table.js [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+            </tbody>
+        </table>
+
+
+
+        )
+    }
+}
+
+export default Table
```
Comment (korczas) [inline src/components/layout/table/Table.js]: - code format
- consider different approach:

```
static propTypes ={
  columns: PropTypes.array.isRequired,
  RowComponent: PropTypes. element.isRequired,
  data: PropTypes.arrayOf(Proptypes.object).isRequired,
}

...

<tbody>
  {this.props.data.map(row => <RowComponent {...row} />}
</tbody>

```

**Review by korczas — CHANGES_REQUESTED** [review]
- format code.
- remove multiple empty rows - you can use empty row to separate some logic, but 3,4,5 empty rows are bad.

App looks nice and clean. After quite a long break with CSS and HTML you're still fresh. Good job.

File: db.json (hunk 2 above)
Comment (monika1127) [reply db.json:62]: dlaczego?

File: src/components/invoices/InvoiceDetails/InvoiceDetails.js (hunk 12 above)
Comment (monika1127) [reply src/components/invoices/InvoiceDetails/InvoiceDetails.js]: ?

**Comment by korczas** [comment]
@monika1127 plz push updates

**Pull request merged**